### PR TITLE
Add missing translations on product edit page

### LINKED ISSUE #4106: Lot's of missing translations on product edit page
## Description
Lots of translations seem to have disappeared in the product editing page :
1.
Capture du 2019-07-31 16-34-04

2. 
Capture du 2019-07-31 16-36-26

3. 
Capture du 2019-07-31 16-37-37

4. 
Capture du 2019-07-31 16-38-59

5.
Capture du 2019-07-31 16-40-15

## Steps to Reproduce

1. On non English speaking instance, go to `admin/products`
2. Choose a product
3. Go on `/edit`

## Severity
S3

## Your Environment

* OFN Platform instance where you discovered the bug, and which version of the software they are using : French v 2.2.1

### PR DESCRIPTION
**Pull request by HugsDaniel**
#### What? Why?

Closes #4106 

Add missing keys for translations and override Spree keys to use our own.

#### What should we test?
Missing translations should now appear. Verify all pages mentioned in the issue are rendering correctly in English.

#### Release notes
Adding missing translations in edit product pages.
Changelog Category: Added | Changed

### CODE REVIEW

**Comment by HugsDaniel** [comment]
In variant edit, when ticking the "On demand" checkbox it shows a missing translation, even though the key is present :thinking: 
Strange thing is that it works in Spanish and French, just not in English... That's what makes the test fail. Any ideas ?

**Comment by luisramos0** [comment]
it a collision @HugsDaniel 
you cannot add "variants: "Variants"" as there is already a variants that has nested keys. that's why variants.infinity doesnt work.

**Comment by luisramos0** [comment]
there are more missing translations in edit variants, you can inspect and search for "missing translations"

also, :variants  is different from .variants, :variants will go to the route. I am not sure admin.products.variants is doing what you need.

**Comment by luisramos0** [comment]
ok, ready for review.

**Comment by luisramos0** [comment]
@HugsDaniel in the last delivery train catch up we decided this was hanging around for quite some time and someone should move it forward.

File: app/views/spree/admin/shared/_edit_resource_links.html.haml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 .form-buttons.filter-actions.actions
-  = button t('update'), 'icon-refresh'
+  = button t(:update), 'icon-refresh'
```
Comment (mkllnk) [inline app/views/spree/admin/shared/_edit_resource_links.html.haml]: This is semantically the same, right? Just better way of writing it.

Comment (luisramos0) [reply app/views/spree/admin/shared/_edit_resource_links.html.haml]: here we move from the key spree.actions.update to the key update.

**Comment by RachL** [comment]
Alright I've staged this on staging UK and nothing looks broken.

Sorry to all new testers to test something as easy as this, but I need this one really badly in production and wanted to make sure this would be included in the next release... this PR is open since September the 5th.... :eyes: let's meeerge it!!

File: app/views/spree/admin/shared/_edit_resource_links.html.haml (hunk 1 above)
Comment (mkllnk) [reply app/views/spree/admin/shared/_edit_resource_links.html.haml]: Really? How does `t('update')` know to look in spree.actions.update? Or do you mean in combination with other commits?

Comment (luisramos0) [reply app/views/spree/admin/shared/_edit_resource_links.html.haml]: 🙈 sorry Maikel, I saw a inexistent dot. yes, this is the same but better looking :-) and less  confusion with the lazylookup.

**Pull request merged**